scripts/e2e_test/e2e_test.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'tmpdir'
+require 'json'
+
+class TestRunner
+  NUM_RUNS_TO_KEEP = 5
+  PROJECT_ROOT = File.expand_path(__dir__ + '/../..') 
+  RUNS_DIR = File.expand_path(__dir__ + '/runs')
+  CYPRESS_DIR = File.join(PROJECT_ROOT, 'packages/cli/cypress')
+
+  E2E_CONFIG = [
+    {
+      name: 'redwood_ts',
+      command: "yarn create redwood-app . --typescript; touch yarn.lock; yarn",
+    },
+    {
+      name: 'redwood_js',
+      command: "yarn create redwood-app .; touch yarn.lock; yarn"
+    },
+    {
+      name: 'next_ts',
+      command: "yarn create next-app . --typescript",
+    },
+    {
+      name: 'next_js',
+      command: "yarn create next-app ."
+    }
+  ];
+
+  def initialize
+    fail "Check that PROJECT_ROOT exists: #{PROJECT_ROOT}" unless Dir.exists?(PROJECT_ROOT)
+    
+    package_json_file = File.join(PROJECT_ROOT, 'package.json')
+    fail "Check that PROJECT_ROOT is the project root: #{PROJECT_ROOT}" unless File.exists?(package_json_file) && 'mailing-monorepo' == JSON::parse(File.read(package_json_file))['name']
+    fail "Check that CYPRESS_DIR exists: #{CYPRESS_DIR}" unless Dir.exists?(CYPRESS_DIR)
+  end
+
+  def build_mailing
+    Dir.chdir(PROJECT_ROOT) do
+      system("yalc add")
+      system("yarn build")
+      system("yalc push")
+    end
+  end
+
+  def run
+    @timestamp_dir = Time.now.strftime("%Y%m%d%H%M%S")
+    
+    # TODO: add a 'skip-build' option to CLI for faster test runs
+    build_mailing
+
+    E2E_CONFIG.each do |config|
+      begin
+        tmp_dir_name = File.join(RUNS_DIR, @timestamp_dir, config[:name])
+
+        puts "⚙️  " * 10
+        puts "Attempting #{config[:name]} in #{tmp_dir_name}"
+        puts "⚙️  " * 10
+
+        FileUtils.mkdir_p(tmp_dir_name)
+
+        Dir.chdir(tmp_dir_name) do
+          # run the bootstrap command
+          system(config[:command])
+
+          ## add mailing to the project
+          system("npx yalc add mailing")
```

Comment:
@psugihara does it make sense that npx is needed here?  i was having some problems without it, but it may be possible to remove

Reply:
yeah, npx there is safer in case node_modules stuff is not in the PATH